hail/python/hail/expr/functions.py

```diff
@@ -2959,8 +2959,10 @@ def len(x) -> Int32Expression:
     """
     if isinstance(x.dtype, ttuple) or isinstance(x.dtype, tstruct):
         return hl.int32(builtins.len(x))
+    elif x.dtype == tstr:
+        return apply_expr(lambda x: StringLength(x), tint32, x)
     else:
-        return x._method("size", tint32)
+        return apply_expr(lambda x: ArrayLen(x), tint32, x)
```

Comment:
does this work with sets? Don't they need to be `ToArray`ed first?

Reply:
I think they do. (Same thing applies for `CollectionExpression.size()` and `.length()` above)

Reply:
Should it?